Decouple computeIntrinsicSizeAndPreferredAspectRatio into two separate functions.

sammygill opened the pull request:
#### 155d07de56e25c9b835595973c4b8c5a31cebb04
<pre>
Decouple computeIntrinsicSizeAndPreferredAspectRatio into two separate functions.
<a href="https://bugs.webkit.org/show_bug.cgi?id=296523">https://bugs.webkit.org/show_bug.cgi?id=296523</a>
<a href="https://rdar.apple.com/problem/156800789">rdar://problem/156800789</a>

Reviewed by Brent Fulgham.

computeIntrinsicSizeAndPreferredAspectRatio currently does two different
things: compute the intrinsic size of the renderer as well as the
preferred aspect ratio. If we instead decouple this into two different
functions, this will improve the readability and maintainability of the
code. For example, any changes that are made to the logic related to
determining the preferred aspect ratio will not need to be concerned
about the other logic related to the renderer&apos;s intrinsic size. In some
cases, you may need the intrinsic size to determine the preferred aspect
ratio, but that will be just a matter of calling the new helper function
whose responsibility will be to give that value to the caller.

* Source/WebCore/rendering/svg/RenderSVGRoot.cpp:
(WebCore::RenderSVGRoot::computeIntrinsicSize const):
(WebCore::RenderSVGRoot::calculateIntrinsicSize const): Deleted.
calculateIntrinsicSize was only being used in two different places:
1.) Inside the constructor of RenderSVGRoot
2.) Directly from the newly added computeIntrinsicSize.
Since these are the only two locations, and is being used in a fairly
straightforward manner in both cases, we probably do not need to keep
both of them around especially since they are both similarly named. It
might end up becoming confusing to determine which one you should call
and when. So I went ahead and moved the logic inside of
calculateIntrinsicSize into the new function and replaced the call
inside the constructor. This required modifying the previous ASSERT we
were using to only trigger when we are inside of render tree layout so
that we can keep using it inside the constructor.

* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGRoot.cpp:
(WebCore::LegacyRenderSVGRoot::computeIntrinsicSize const):
(WebCore::LegacyRenderSVGRoot::calculateIntrinsicSize const): Deleted.
Same as RenderSVGRoot.cpp.

Canonical link: <a href="https://commits.webkit.org/298315@main">https://commits.webkit.org/298315@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1b9ff84d91b168e43c27e101dbffd9add2881dd3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/114636 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/34382 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/24845 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/120802 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/65358 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/01aa4878-49b4-44fd-85ad-a8a6964a0169) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/116525 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/35010 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/42942 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/87142 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/42043 "") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/f32bad56-8d6c-4def-86f4-d88495d3431b) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/117584 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/27894 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/102953 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/67530 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/27081 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/21077 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/64479 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/97274 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/21187 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/124004 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/41648 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/31102 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/95952 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/42025 "Built successfully") | [❌ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/99138 "") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/95733 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24440 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/40899 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/18744 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/37747 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/41526 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/47038 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/41112 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/44419 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/42862 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/134bc87c956ed74b9f76c6e72782305d00fb1c0c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48529

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d5ef3693a987bdd1154a780585d50a44628720bc. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48529

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/0b4e15a4c19ef03a3387d443696d79c2004b2836. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48529

sammygill commented on Source/WebCore/rendering/RenderReplaced.cpp:
```diff
@@ -567,17 +577,17 @@ std::pair<FloatSize, FloatSize> RenderReplaced::computeIntrinsicSizeAndPreferred
+        return preferredAspectRatio;
 
-    return { intrinsicSize, { intrinsicSize.width(), intrinsicSize.height() } };
+    return { intrinsicSize.width(), intrinsicSize.height() };
```
Good catch! Yeah this is probably a bit wasteful and I don't see a reason why we would need to do that

sammygill commented on Source/WebCore/rendering/svg/legacy/LegacyRenderSVGRoot.cpp:
```diff
@@ -93,16 +93,20 @@ FloatSize LegacyRenderSVGRoot::calculateIntrinsicSize() const
+FloatSize LegacyRenderSVGRoot::computeIntrinsicSize() const
 {
     ASSERT(!shouldApplySizeContainment());
+    return calculateIntrinsicSize();
```
Makes sense I went ahead and got rid of `calculateIntrinsicSize` and stuck with `computeIntrinsicSize`

brentfulgham reviewed (approved):
Sorry I didn't review this after you made the changes. r=me!

sammygill committed:
Decouple computeIntrinsicSizeAndPreferredAspectRatio into two separate functions.
https://bugs.webkit.org/show_bug.cgi?id=296523
rdar://problem/156800789

Reviewed by Brent Fulgham.

computeIntrinsicSizeAndPreferredAspectRatio currently does two different
things: compute the intrinsic size of the renderer as well as the
preferred aspect ratio. If we instead decouple this into two different
functions, this will improve the readability and maintainability of the
code. For example, any changes that are made to the logic related to
determining the preferred aspect ratio will not need to be concerned
about the other logic related to the renderer's intrinsic size. In some
cases, you may need the intrinsic size to determine the preferred aspect
ratio, but that will be just a matter of calling the new helper function
whose responsibility will be to give that value to the caller.

* Source/WebCore/rendering/svg/RenderSVGRoot.cpp:
(WebCore::RenderSVGRoot::computeIntrinsicSize const):
(WebCore::RenderSVGRoot::calculateIntrinsicSize const): Deleted.
calculateIntrinsicSize was only being used in two different places:
1.) Inside the constructor of RenderSVGRoot
2.) Directly from the newly added computeIntrinsicSize.
Since these are the only two locations, and is being used in a fairly
straightforward manner in both cases, we probably do not need to keep
both of them around especially since they are both similarly named. It
might end up becoming confusing to determine which one you should call
and when. So I went ahead and moved the logic inside of
calculateIntrinsicSize into the new function and replaced the call
inside the constructor. This required modifying the previous ASSERT we
were using to only trigger when we are inside of render tree layout so
that we can keep using it inside the constructor.

* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGRoot.cpp:
(WebCore::LegacyRenderSVGRoot::computeIntrinsicSize const):
(WebCore::LegacyRenderSVGRoot::calculateIntrinsicSize const): Deleted.
Same as RenderSVGRoot.cpp.

Canonical link: https://commits.webkit.org/298315@main
